tests/nodes/tests_Literal.cpp

```diff
@@ -457,3 +459,237 @@ TEST_CASE("Literal - casting") {
+        CHECK("abracadabra"_xsd_string .as_regex_matches("bra"_xsd_string).ebv());
+        CHECK("abracadabra"_xsd_string .as_regex_matches("^a.*a$"_xsd_string).ebv());
+        CHECK("abracadabra"_xsd_string .as_regex_matches("^bra"_xsd_string).ebv() == util::TriBool::False);
+
+        std::string_view const poem = "<poem author=\"Wilhelm Busch\">\n"
+                                      "Kaum hat dies der Hahn gesehen,\n"
+                                      "Fängt er auch schon an zu krähen:\n"
+                                      "Kikeriki! Kikikerikih!!\n"
+                                      "Tak, tak, tak! - da kommen sie.\n"
+                                      "</poem>";
+        Literal const poem_lit{poem};
+
+        CHECK(poem_lit.as_regex_matches("Kaum.*krähen"_xsd_string).ebv() == util::TriBool::False);
+        //CHECK(poem_lit.regex_match("^Kaum.*gesehen,$"_xsd_string, "m"_xsd_string).ebv()); TODO: support multiline flag
+        CHECK(poem_lit.as_regex_matches("^Kaum.*gesehen,$"_xsd_string).ebv() == util::TriBool::False);
+        CHECK(poem_lit.as_regex_matches("kiki"_xsd_string, "i"_xsd_string).ebv());
+
+        // check lang tag behaviour
+        CHECK(Literal{"abcd", "en"}.as_regex_matches("b"_xsd_string).ebv());
+        CHECK(Literal{"abcd", "en"}.as_regex_matches(Literal{"b", "en"}).ebv());
+        CHECK(Literal{"abcd", "en"}.as_regex_matches(Literal{"b", "fr"}).null());
+    }
+
+    SUBCASE("regex_replace") {
+        // from https://www.w3.org/TR/sparql11-query/#func-replace
+        CHECK("abcd"_xsd_string .regex_replace("b"_xsd_string, "Z"_xsd_string) == "aZcd"_xsd_string);
+        CHECK("abab"_xsd_string .regex_replace("B"_xsd_string, "Z"_xsd_string, "i"_xsd_string) == "aZaZ"_xsd_string);
+        CHECK("abab"_xsd_string .regex_replace("B."_xsd_string, "Z"_xsd_string, "i"_xsd_string) == "aZb"_xsd_string);
+
+        // from https://www.w3.org/TR/xpath-functions/#func-replace
+        CHECK("abracadabra"_xsd_string .regex_replace("bra"_xsd_string, "*"_xsd_string) == "a*cada*"_xsd_string);
+        CHECK("abracadabra"_xsd_string .regex_replace("a.*a"_xsd_string, "*"_xsd_string) == "*"_xsd_string);
+        CHECK("abracadabra"_xsd_string .regex_replace("a.*?a"_xsd_string, "*"_xsd_string) == "*c*bra"_xsd_string);
+        CHECK("abracadabra"_xsd_string .regex_replace("a"_xsd_string, ""_xsd_string) == "brcdbr"_xsd_string);
+        CHECK("abracadabra"_xsd_string .regex_replace("a(.)"_xsd_string, "a$1$1"_xsd_string) == "abbraccaddabbra"_xsd_string);
+        CHECK("AAAA"_xsd_string .regex_replace("A+"_xsd_string, "b"_xsd_string) == "b"_xsd_string);
+        CHECK("AAAA"_xsd_string .regex_replace("A+?"_xsd_string, "b"_xsd_string) == "bbbb"_xsd_string);
+        CHECK("darted"_xsd_string .regex_replace("^(.*?)d(.*)$"_xsd_string, "$1c$2"_xsd_string) == "carted"_xsd_string);
+
+        // 'The expression fn:replace("abracadabra", ".*?", "$1") raises an error, because the pattern matches the zero-length string'
```

Comment:
What exactly are the problems here? If it turns out to be complex, I would suggest to create an issue, fix the other things in ths PR and merge.

Reply:
This is basically behaviour coming from RE2 and I'm not sure how to fix it from the outside.

Reply:
opened an issue #132